### PR TITLE
image generation main process checks

### PR DESCRIPTION
**Pull request by williamberman**

### CODE REVIEW

File: examples/text_to_image/train_text_to_image_lora.py [hunk 1]
```diff
@@ -800,20 +800,19 @@ def collate_fn(examples):
...
+                        np_images = np.stack([np.asarray(img) for img in images])
+                        tracker.writer.add_images("validation", np_images, epoch, dataformats="NHWC")
+                    if tracker.name == "wandb":
+                        tracker.log(
+                            {
+                                "validation": [
+                                    wandb.Image(image, caption=f"{i}: {args.validation_prompt}")
+                                    for i, image in enumerate(images)
+                                ]
+                            }
```
Comment (williamberman) [inline examples/text_to_image/train_text_to_image_lora.py:814]: this is already gated on if this is the main process on line 779 https://github.com/huggingface/diffusers/pull/2631/files#diff-911b29c0306dfe271bf7f6c3070c94623b2141e3f02a37c2f28cb726bba5ee3eL779

**Review by yiyixuxu — APPROVED** [review]
Thanks!

**Review by patrickvonplaten — APPROVED** [review]
Thanks a lot!

**Comment by sayakpaul** [comment]
I'd like to review this today for my own understanding. But feel free to merge if need be before that.

**Review by sayakpaul — APPROVED** [review]
Thanks for streamlining this!

**Review by pcuenca — APPROVED** [review]
Awesome, thanks!

**Pull request merged**